packages/api/routes/apikey/deleteKeyMiddleware.js

```diff
@@ -0,0 +1,12 @@
+const db_apikey = require("../../lib/db.apikey");
+
+// Return a function for deleting an API key passed as a parameter.
+module.exports = function deleteKeyMiddleware() {
+  return async (req, res, next) => {
+    const apikey = await db_apikey.attach();
+    apikey.deleteKey(req.params.hashedKey).then(() => {
+      res.send("Key deleted");
+      next();
+    });
+  };
+};
```

Comment:
This needs to handle error conditions.

 - database connection fails
 - hashed key parameter is missing
 - hashed key parameter is empty string
 - (optional) hashed key parameter is wrong length.  keys are hashed with sha256 so they're always 64 characters
 - (optional) hashed key has invalid characters.  sha256 produces hex characters, anything else is invalid
 - hashed key did not exist in the database (you'll need to update db.apikey.deleteKey to return a value first)

Reply:
I thought more about the error conditions this afternoon and I think I was too hasty.  There are other parts of SPAship that _dont_ have super robust error checking, so your code would fit right in.  It's okay if you want to wait until after MVP to add really thorough handling of error conditions.  I'll leave it up to you @sayak-sarkar.  (I'm leaving this same message on my other error condition comment.)